src/solvers/smt2/smt2_conv.cpp

```diff
@@ -846,7 +846,10 @@ void smt2_convt::convert_address_of_rec(
       struct_op_type.id() == ID_struct || struct_op_type.id() == ID_struct_tag,
       "member expression operand shall have struct type");
 
-    const struct_typet &struct_type = to_struct_type(ns.follow(struct_op_type));
+    const struct_typet &struct_type =
+      struct_op_type.id() == ID_struct_tag
+        ? ns.follow_tag(to_struct_tag_type(struct_op_type))
+        : to_struct_type(struct_op_type);
```

Comment:
I'm afraid I don't think this is the case: symex does resolve tag types, so the back-end may well see expanded `struct_typet` rather than `struct_tag_typet`.